src/components/TokenForm/TokenForm.tsx

```diff
@@ -210,18 +206,23 @@ const TokenForm: FC<TokenFormProps> = ({
             className="col-span-full mt-3 grid"
             disabled={!form.formState.isValid}
             isLoading={
-              isLoadingInputToken ||
-              isLoadingInputTokenBalanceOrShare ||
-              isLoadingTransaction
+              !isError &&
+              (isLoadingInputToken ||
+                isLoadingInputTokenBalanceOrShare ||
+                isLoadingTransaction)
             }
             type="submit"
           >
-            {form.formState.isDirty
+            {isError
+              ? "Error preparing transaction"
+              : form.formState.isDirty
               ? form.formState.isValid
                 ? submitText
-                : form.formState.errors.amountIn === undefined
-                ? "Enter an amount"
-                : form.formState.errors.amountIn.message ?? "Unknown error"
+                : form.formState.errors.amountIn
+                ? `Insufficient ${inputToken?.symbol ?? ""} ${
```

Comment:
Moving this messaging out of the `validate` callback so that it will always reflect the token the user has selected.